features/project/source/browse_files.feature

```diff
@@ -33,6 +33,13 @@ Feature: Project Source Browse Files
     And I click on "Commit Changes"
     Then I am redirected to the new file
     And I should see its new content
+    
+  @javascript
+  Scenario: I can upload file and commit
+    Given I click on "new file" link in repo
+    Then I can see new file page
+    And I can see "upload existing one"
+    And I click on "upload existing one"
```

Comment:
Are you able to verify the upload contents? For example, you may be able to download the file via the raw controller.